Uyanda.Coffee.Persistence/Accessors/BeverageAccessor.cs

```diff
@@ -67,6 +67,56 @@ public async Task<InvoiceModel> PurchaseAsync(IEnumerable<LineItemModel> lineIte
             return ToModel(invoice);
         }
 
+        public async Task<BeverageSizeCostModel[]> AddBeverageSizeCostAsync(BeverageSizeCostModel price)
+        {
+          
+            var beverageDictionary = await localDbContext.Beverages.AsNoTracking()
+                .Select(c => new { c.Id, c.Name }).ToDictionaryAsync(c=>c.Id, c=>c.Name);
+
+            if (!beverageDictionary.ContainsValue(price.Beverage.Name))
+            {
+                localDbContext.BeverageCost.Add(ToEntity(price));
+
+                await localDbContext.SaveChangesAsync();
+
+                var result = await localDbContext.BeverageCost.AsNoTracking().Select(c => c).ToArrayAsync();
+
+                return result.Select(ToModel).ToArray();
+            }
+
+            var beverageSizeDictionary = localDbContext.BeverageCost.AsNoTracking()
+                .Where(x => x.Beverage.Name == price.Beverage.Name)
+                .Select(c => new { c.BeverageSizeId, c.BeverageSize.Name }).ToDictionary(b => b.BeverageSizeId, b => b.Name);
+
+            if(beverageSizeDictionary.ContainsKey(price.BeverageSizeId))
+            {
+
+                var record = await localDbContext.BeverageCost
+                    .SingleOrDefaultAsync(c => c.BeverageSizeId == price.BeverageSizeId && c.Beverage.Name == price.Beverage.Name);
```

Comment:
Checking the Id only suffices. . . Also, if a name was edited, then you can never update it with this check existing

Reply:
when I remove the name comparisons, or even the Id comparisons. This query throws an exception.